application.go

```diff
@@ -102,3 +125,59 @@ func (impl *applicationImpl) CreateApplication(ctx context.Context, app *Applica
 	}
 	return &createdApp, nil
 }
+
+func (impl *applicationImpl) UpdateApplication(ctx context.Context, app *Application, opts *ResourceUpdateOptions) (*Application, error) {
```

Comment:
do we need PatchApplication method?

Reply:
A `PatchApplication is not easy to support since it's difficult for users to specify only a few fields for the `Application`. We need to consider the `Application` type design carefully. Or empty fields will be set the zero value.